CMakeLists.txt

```diff
@@ -260,21 +240,40 @@ if(PORTABLE)
   if(FORCE_SSE42 AND NOT MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2 -mpclmul")
   endif()
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64le")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=power8")
+    else()
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=power7")
+    endif()
+  endif()
 else()
   if(MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /arch:AVX2")
   else()
-    if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64" AND NOT HAS_ARMV8_CRC)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+    if(NOT CMAKE_CXX_FLAGS MATCHES "m(cpu|tune|arch)")
+      if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
+        # Power doesn't have a march option - https://gcc.gnu.org/onlinedocs/gcc/RS_002f6000-and-PowerPC-Options.html
+        # -mcpu activates all available options
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=native")
+      else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+      endif()
     endif()
   endif()
 endif()
 
-include(CheckCXXSourceCompiles)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
+  include(CheckCXXCompilerFlag)
+  SET(POWER_CRC32_FLAGS "-mcrypto -mpower8-vector -maltivec -mvsx")
```

Comment:
I am wondering about the `power8` part here. Isn't that CPI specific. What about Power7 and Power9 as mentioned above? I am struggling to follow the flow a bit, but is this also set when `PORTABLE=1` and if so, should it be?

Reply:
I put this in the commit message but it probably should be a comment here too.

We include this when the compile flags enable it to be build.
For example on POWER7 the optimized crc32 will be built,
however it won't be used because of the runtime detection.
If the executable is moved to a POWER8 big endian it will
work correctly.`